pkgs/sentry/default.nix

```diff
@@ -0,0 +1,72 @@
+{ fetchFromGitHub
+, fetchpatch
+, lib
+, libtool
+, libxml2
+, libxslt
+, postgresql
+, python
+, xmlsec
+, zlib
+}:
+
+let
+
+  iohkMkPythonApplication = { python, overrides, ... }@attrs:
+    let
+      specialAttrs = [
+        "overrides"
+      ];
+      passedAttrs = builtins.removeAttrs attrs specialAttrs;
+
+      deps = map (depName: py.pkgs."${depName}") (builtins.attrNames (overrides {} {}));
+
+      packageOverrides = lib.foldr lib.composeExtensions (self: super: {}) [overrides];
+
+      py = python.override { inherit packageOverrides; self = py; };
+    in
+      python.pkgs.buildPythonApplication (
+        passedAttrs // {
+          propagatedBuildInputs = (attrs.propagatedBuildInputs or []) ++ deps;
+        }
+      );
+in
+
+(iohkMkPythonApplication rec {
+  pname   = "sentry";
+  version = "10.0.0";
+  format = "wheel";
+
+  src = python.pkgs.fetchPypi {
+    inherit pname version format;
+    sha256 = "2695ed1cf11e5afee92de04dc1868de9f4c49d5876c34579c935d9d6cf9fbb03";
+    python = "py27";
+  };
+
+  # python = pkgs.python;
+  inherit python;
+  overrides = import ./overrides.nix {
```

Comment:
Is this file `overrides.nix` automatically generated? It seems like it might be difficult to maintain when bumping nixpkgs or then sentry version. Perhaps one of the _2nix_ tools for python would make things easier?

Reply:
It wasn't automatically generated. I had no initial success with pypi2nix and the alternatives, and my knowledge of the python nixpkgs infra wasn't enough to realise these issues were surmountable, so I (mostly) hand-rolled the overrides for Sentry and Snuba. It took a long time but I learnt a lot. Enough to realize that I can use pypi2nix and so I've gone ahead and done that for Snuba and Sentry. Thanks :)